src/bin/ion/commands/generate/generator.rs

```diff
@@ -300,10 +337,16 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
         schema_id: &str,
     ) -> CodeGenResult<()> {
         let schema = schema_system.load_isl_schema(schema_id).unwrap();
-        self.generate(schema)
+        // For a single schema file, need to generate code for header and inline imports
+        self.generate(schema, true, schema_system)
     }
 
-    fn generate(&mut self, schema: IslSchema) -> CodeGenResult<()> {
+    fn generate(
+        &mut self,
+        schema: IslSchema,
+        generate_imports: bool, // if this is set as false, then it skips generation of header imports and inline imports
```

Comment:
Why would you ever want to skip generating imported types? If your schema imports something you have to generate that or else you'll end up with e.g. a `ClassNotFoundException` when your generated `Foo` class has a field of type `Bar` and `Bar` was not generated.

Reply:
If the use provided an authority(s) `-d` option then all the schema files inside of that directory would get generated. So then if we again generate the same schema file/ schema type with an import then it generates error for specifically Rust code generation as it creates every data model into the same module.

So this property `generate_imports` allows generating import when a single schema file is being generated with CLI option `--schema`. Otherwise, when entire authority is being generated this will be set as false, which skips generating imports as they would already be generated while traversing through each file in the authority.

Reply:
I think we should drop the `--schema` option (for now) in order to simplify things. In terms of use cases we should eventually support, I think the two most important, by far, are
1. Generated everything in a directory/authority
2. Given one or more authorities, allow the user to specify one or more types they would like to generate, and then generate those types _and_ all the dependencies for those types.

Reply:
Removed `--schema` option, user can still specify an authority with a single schema file which works the same way.